core/src/com/unciv/logic/city/CityConstructions.kt

```diff
@@ -71,6 +71,13 @@ class CityConstructions {
         val stats = Stats()
         for (building in getBuiltBuildings())
             stats.add(building.getStats(cityInfo.civInfo))
+
+        for (unique in builtBuildingUniqueMap.getAllUniques()) when (unique.placeholderText) {
+            "[] Per [] Population in this city" -> stats.add(Stats.parse(unique.params[0]).times(cityInfo.population.population / unique.params[1].toFloat()))
```

Comment:
> // Doing this through the hashmap is nicer code but is SUPER INEFFICIENT!

```
    operator fun times(number: Float): Stats {
        val hashMap = toHashMap()
        for (stat in Stat.values()) hashMap[stat] = number * hashMap[stat]!!
        return Stats(hashMap)
    }
```
Is this example of using hashmap is also bad?

Reply:
It's also inefficient but that's a much rarer use-case, and so it doesn't cause any problems.
The other function came us as problematic when I was doing memory profiling at I changed it